R/filter_state_api_NEW.R

```diff
@@ -0,0 +1,442 @@
+#' This is always a logical flag that defaults to FALSE.
+#' In a `FilterState` instantiated with `fixed = TRUE` **none** of the state features can be changed.
+#'
+#' `filter_var` creates a `teal_slice` object, which specifies a filter for a single variable,
+#' passed to and resolved by `FilterState` objects.
+#' `filter_settings` collates multiple `teal_slice` objects into `teal_slices`,
+#' a complete filter specification. This is used by all classes above `FilterState`
+#' as well as `filter_panel_api` wrapper functions.
+#' `teal_slices` also specifies which variables cannot be filtered
+#' and how observations are tallied, which is resolved by `FilterStates`.
+#'
+#' @param dataname `character(1)` name of data set
+#' @param varname `character(1)` name of variable
+#' @param choices vector specifying allowed choices;
+#'                possibly a subset of values in data;
+#'                type and size depends on variable type
+#' @param selected vector specifying selection;
+#'                 type and size depends on variable type
+#' @param keep_na `logical(0-1)` optional logical flag specifying whether to keep missing values
+#' @param keep_inf `logical(0-1)` optional logical flag specifying whether to keep infinite values
+#' @param fixed `logical(1)` logical flag specifying whether to fix this filter state (i.e. forbid setting state)
+#' @param ... for `filter_var` any number of additional features given as `name:value` pairs\cr
+#'            for `filter_settings` any number of `teal_slice` objects
+#' @param exclude `named list` of `character` vectors where list names match names of data sets
+#'                 and vector elements match variable names in respective data sets;
+#'                 specifies which variables are not allowed to be filtered
+#' @param count_type `character(1)` string specifying how observations are tallied by these filter states
+#'
+#' @return
+#' `filter_var` returns object of class `teal_slice`, which is a named list.
+#' `filter_settings` returns object of class `teal_slices`, which is an unnamed list of `teal_slice` objects.
+#'
+#' @examples
+#' filter_1 <- filter_var("dataname1", "varname1", letters, "b", FALSE, extra1 = "extraone")
+#' filter_2 <- filter_var("dataname1", "varname2", 1:10, 2, TRUE, FALSE, extra2 = "extratwo")
+#' filter_3 <- filter_var("dataname2", "varname3", 1:10/10, 0.2, TRUE, FALSE,
+#'                        extra1 = "extraone", extra2 = "extratwo")
+#'
+#' all_filters <- filter_settings(
+#'   filter_1,
+#'   filter_2,
+#'   filter_3,
+#'   exclude = list(
+#'     "dataname1" = "varname2"
+#'   )
+#' )
+#'
+#' extract_by_feat(all_filters, "dataname", "dataname1")
+#' extract_feat(all_filters, "dataname")
+#' extract_fun(all_filters, dataname == "dataname2")
+#' extract_fun_s(all_filters, 'dataname == "dataname2"')
+#' x <- "dataname2"
+#' extract_fun_s(all_filters, sprintf('dataname == "%s"', x))
+#'
+#'NULL
+
+
+#' @export
+#' @rdname new_api
```

Comment:
Also `@name` tag is missing in the docs of the `NULL` above
```suggestion
#' @rdname teal_slice_api
```

Reply:
How about just `teal_slice`? Too confusing with package name?

Reply:
teal_slice is 👍